warehouse/admin/views/projects.py

```diff
@@ -201,3 +202,32 @@ def journals_list(project, request):
     )
 
     return {"journals": journals, "project": project, "query": q}
+
+
+@view_config(
+    route_name="admin.project.set_upload_limit",
+    permission="admin",
+    request_method="POST",
+    uses_session=True,
+    require_methods=False,
+)
+def set_upload_limit(project, request):
+    upload_limit = request.POST.get("upload_limit", "")
+
+    # If the upload limit is an empty string or othrwise falsy, just set the
+    # limit to None, indicating the default limit.
+    if not upload_limit:
+        upload_limit = None
+
+    # Update the project's upload limit.
+    project.upload_limit = int(upload_limit)
```

Comment:
This isn't going to work if ``upload_limit`` is ``None``. Maybe do

```python

if not upload_limit:
    project.upload_limit = None
else:
    try:
        project.upload_limit = int(upload_limit)
    except ValueError:
        raise HTTPBadRequest(f"Invalid value for upload_limit: {upload_limit}) from None
```

That should cover all of the cases I think?

Reply:
Yep, done - without this SQLAlchemy will raise an error, but this is more explicit and raises a proper HTTP error.

Reply:
Yea, SQLAlchemy errors get bubbled up into Sentry and alerted whereas HTTP errors don't. So when there is an expected error we try to catch it and turn it into HTTP errors to keep sentry cleaner. Not really a super big deal here because I doubt the 2 or 3 admins are likely to start throwing a bunch of random garbage into fields, but whatever!

Reply:
Better safe than alerted.